### DISCUSSION TITLE
Controlling individual pins

### DISCUSSION THREAD

**Discussion by Seggan** [mattjlewis/diozero#74] [Q&A] [answered]
Is there a way to turn individual pins on/off? I thought maybe this would be it: `RaspberryPiBoardInfoProvider.lookupByRevision(LocalSystemInfo.getInstance().getRevision()).createMmapGpio().gpioWrite()`

**Comment by mattjlewis** [comment, marked as answer]
Create an instance of `com.diozero.api.DigitalOutputDevice` ([Javadoc](https://www.javadoc.io/doc/com.diozero/diozero-core/latest/com/diozero/api/DigitalOutputDevice.html)). You then have access to `on()` / `off()` / `toggle()` / `setValue(boolean)` methods.

**Reply by Seggan** [reply]
but what if i want to read and write from the same pin? im creating a class for a 4x4 membrane switch and i need to read value, switch pin modes, and then write

**Reply by Seggan** [reply]
nvm found `DigitalInputOutputDevice`

**Reply by mattjlewis** [reply]
4x4 membrane keypads should already be supported - see: https://github.com/mattjlewis/diozero/blob/main/diozero-core/src/main/java/com/diozero/devices/Keypad.java

**Reply by Seggan** [reply]
Thanks, I did not notice
